### PR TITLE
extend-ui-filter-with-message-header

### PR DESCRIPTION
**Pull request by luethiandrin**
image

---

image

### CODE REVIEW

**Comment by luethiandrin** [comment]
The scenario-detail view doesn load the filter buttons on init. It works if you press F5 after the page has loaded. Needs to be fixed @bbortt.

File: simulator-ui/src/main/resources/static/angular.json [hunk 1]
```diff
@@ -29,10 +29,12 @@
             "styles": [
               "./src/assets/css/main.css",
               "./node_modules/bootstrap/dist/css/bootstrap.min.css",
-              "./node_modules/font-awesome/css/font-awesome.min.css"
+              "./node_modules/font-awesome/css/font-awesome.min.css",
+              "./node_modules/@angular/material/prebuilt-themes/deeppurple-amber.css"
             ],
             "scripts": [
               "node_modules/bootstrap/dist/js/bootstrap.bundle.js"
+
```
Comment (bbortt) [inline simulator-ui/src/main/resources/static/angular.json:37]: remove newline, please

File: simulator-ui/src/main/resources/static/package.json [hunk 2]
```diff
@@ -15,10 +15,12 @@
   },
   "private": true,
   "dependencies": {
+    "@angular/animations": "^13.0.2",
     "@angular/common": "^13.0.2",
     "@angular/compiler": "^13.0.2",
     "@angular/core": "^13.0.2",
     "@angular/forms": "^13.0.2",
+    "@angular/material": "^13.0.2",
```
Comment (bbortt) [inline simulator-ui/src/main/resources/static/package.json:23]: it's interesting to have bootstrap and material on the classpath 😅

File: simulator-ui/src/main/resources/static/src/app/components/activity/activity.css [hunk 3]
```diff
@@ -5,4 +5,4 @@
 
 .panel-heading div span{
     margin-left:5px;
-}
+}
```
Comment (bbortt) [inline simulator-ui/src/main/resources/static/src/app/components/activity/activity.css:8]: newline at the end of the file is good manner.

File: simulator-ui/src/main/resources/static/src/app/model/scenario.ts [hunk 4]
```diff
@@ -77,4 +77,4 @@ export class MessageHeader {
         public name: string,
         public value: string) {
     }
-}
+}
```
Comment (bbortt) [inline simulator-ui/src/main/resources/static/src/app/model/scenario.ts:80]: don't touch this file 😉

File: simulator-ui/src/main/resources/static/src/assets/css/filter-section.css [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    background-color:#000!important;
+}
+
+::ng-deep .mat-calendar-body-cell:not(.mat-calendar-body-disabled):hover > .mat-calendar-body-cell-content:not(.mat-calendar-body-selected):not(.mat-calendar-body-comparison-identical) {
+    background-color:#5bc0de;
+}
+
+::ng-deep .mat-calendar-body-today:not(.mat-calendar-body-selected):not(.mat-calendar-body-comparison-identical) {
+    border-color:#31708f;
+    background-color:#5bc0de!important;
```
Comment (bbortt) [inline simulator-ui/src/main/resources/static/src/assets/css/filter-section.css:108]: this file looks unformatted. as mentioned [here](https://github.com/citrusframework/citrus-simulator/pull/111#pullrequestreview-824913773), try avoiding `!important`..

**Review by bbortt — CHANGES_REQUESTED** [review]

File: simulator-ui/src/main/resources/static/package.json (hunk 2 above)
Comment (luethiandrin) [reply simulator-ui/src/main/resources/static/package.json:23]: Why?

Comment (bbortt) [reply simulator-ui/src/main/resources/static/package.json:23]: well, two different css frameworks..

**Comment by bbortt** [comment]
I merged this into the intermediate branch straight away. will give a final review there :) thanks very much.

**Pull request merged**